entity-table/index.js

```diff
@@ -182,38 +182,57 @@ module.exports = {
 			);
 		});
 	},
-	tableOldNewProcessor: function(options) {
+	tableOldNewProcessor: function(optionsIn) {
+		let options = Object.assign({ kinesisBatchLimit: 200}, optionsIn);
 		let self = this;
 		return function(event, context, callback) {
 			let streams = {};
+			let batchSize = event.Records.length;
+			let filePrefix = event.Records[0].eventID;
+			logger.info(`batchSize=${batchSize}`);
+			let localReadOpts = Object.assign({
+				prefix: filePrefix,
+				toCheckpoint: { force: true}
+			}, options.read);
+			if (batchSize >= options.kinesisBatchLimit) {
+				logger.info(`using S3 for batchSize of ${batchSize}`);
+				localReadOpts.useS3 = true;
+			}
+			else {
+				logger.info(`using Kinesis for batchSize of ${batchSize}`);
+				localReadOpts.useS3 = false;
+			}
+			
 			let index = 0;
 			let defaultQueue = options.defaultQueue || "Unknown";
 			let resourcePrefix = sanitizePrefix(options.resourcePrefix);
 			let resourceSuffix = options.eventSuffix || "_table_changes";
-			let getStream = id => {
-				if (!(id in streams)) {
-					streams[id] = ls.pipeline(leo.write(id), ls.toCheckpoint({
-						force: true
-					}));
+			let getStream = (id, queue) => {
+				let key = `${id}/${queue}`;
```

Comment:
What is the reason for splitting on bot and queue.  Will it be different values in a single run?  ever be

Reply:
@zirkerc I don't know. This was just taking the code that you and I came up with when we had to use S3 for various entity tables. I don't remember anymore why we did it that way, just that we did.